matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/auth/registration/DefaultRegistrationWizard.kt

```diff
@@ -187,22 +193,51 @@ internal class DefaultRegistrationWizard(
         return performRegistrationRequest(params)
     }
 
-    private suspend fun performRegistrationRequest(registrationParams: RegistrationParams,
-                                                   delayMillis: Long = 0): RegistrationResult {
+    override suspend fun registrationOther(
+            authParams: JsonDict
+    ): RegistrationResult {
+        val safeSession = pendingSessionData.currentSession
+                ?: throw IllegalStateException("developer error, call createAccount() method first")
+
+        val mutableParams = authParams.toMutableMap()
+        mutableParams["session"] = safeSession
```

Comment:
Ah you are already doing that anyway! So what's the point to expose the `sessionId` to the app?

Reply:
Added this in case no one of existing methods in RegistrationWizard is not suitable for sdk user. And this will allow to build custom stage implementation on app side

Reply:
`registrationCustom` should cover every other cases, no?

Reply:
Added just in case it doesn’t. But actually it should be enough for all cases. Ok, will remove it.